base/essentials.jl

```diff
@@ -861,9 +861,9 @@ end
 macro inbounds(blk)
     return Expr(:block,
         Expr(:inbounds, true),
-        Expr(:local, Expr(:(=), :val, esc(blk))),
+        Expr(:local, Expr(:(=), :val_49f6338b, esc(blk))),
```

Comment:
`gensym`? But as was pointed out in the original issue, the macro is fine as written, the current behavior is a bug. Maybe leave a comment that effect?

Reply:
`gensym` is not defined during bootstrapping so this seemed easiest.

Reply:
ok

Reply:
This is off-topic, but I wonder if reserved words might be useful in these cases. cf. (PR #51215)
Of course, we still need to be careful to avoid collisions in `Base` and stdlibs.